src/libraries/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -1923,6 +1949,13 @@ public void CreateInstance_NotSupportedType_ThrowsNotSupportedException(Type ele
             Assert.Throws<NotSupportedException>(() => Array.CreateInstance(elementType, new int[1]));
             Assert.Throws<NotSupportedException>(() => Array.CreateInstance(elementType, new long[1]));
             Assert.Throws<NotSupportedException>(() => Array.CreateInstance(elementType, new int[1], new int[1]));
+
+            Assert.Throws<ArgumentException>("arrayType", () => Array.CreateInstanceFromArrayType(elementType, 0));
```

Comment:
```suggestion
            Assert.Throws<ArgumentException>("arrayType", () => Array.CreateInstanceFromArrayType(elementType.MakeArrayType(), 0));
```
These tests need to create the array type from the elementType.

I expect that some of them going to fail since the implementation does not have proper error handling. It is a bug in the implementation that should be fixed.